cmd/nerdctl/pull.go

```diff
@@ -92,6 +100,46 @@ func pullAction(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	if isVerify, err := cmd.Flags().GetString("verify"); err == nil && isVerify == "cosign" {
+		cosignExecutable, err := exec.LookPath("cosign")
+		if err != nil {
+			logrus.WithError(err).Error("cosign executable not found in path $PATH")
+			logrus.Info("you might consider installing cosign from: https://docs.sigstore.dev/cosign/installation")
+			return err
+		}
+
+		cosignCmd := exec.Command(cosignExecutable, []string{"verify"}...)
+		cosignCmd.Env = os.Environ()
+
+		keyRef, err := cmd.Flags().GetString("cosign-key")
+		if err != nil {
+			return err
+		}
+
+		if keyRef != "" {
+			cosignCmd.Args = append(cosignCmd.Args, "--key", keyRef)
+		} else {
+			cosignCmd.Env = append(cosignCmd.Env, "COSIGN_EXPERIMENTAL=true")
+		}
+
+		cosignCmd.Args = append(cosignCmd.Args, rawRef)
```

Comment:
This is racy (https://github.com/sigstore/cosign/issues/648), so at least we have to print a warning when `rawRef` does not contain the explicit digest value.

Ideally, we should resolve the digest before shelling out `cosign`.

Reply:
that's so true, thanks for warning, done.

Reply:
The latest revision seems to have this race again